src/app/core/public/api/impl/Circuit.ts

```diff
@@ -155,7 +166,35 @@ export abstract class CircuitImpl implements Circuit {
         //  For now, ignore the `space`, and ignore any non-Component
         //   objects that are selected
         //  From these components, average their positions
-        throw new Error("Method not implemented.");
+
+        var allComponents = this.selections.get();
+
+        var xPosition = 0;
+        var yPosition = 0;
+        var comp = null;
```

Comment:
nit: Use `undefined` instead of `null`
In this case, you don't have to initialize `comp` as anything here. You can just create the variable when it is used below, like
```typescript
const comp = this.getComponent(allComponents[0]);
```
and
```typescript
const comp = this.getComponent(id);
```